canonical-web-and-design/maas-ui
# Add Cypress with basic login tests.

## Done
- Added cypress npm package to proxy workspace
- Added start-server-and-test npm package to proxy workspace, which waits until all servers are running before starting cypress tests
- Added basic login tests
- Added cypress testing to CI

## QA
- Install new packages (`yarn`)
- Start maas-ui (`yarn serve`)
- Once it's up and running, open a new terminal and run `yarn cypress-open`
- The Cypress test runner should pop up. Click "Run all specs" in the top-right
- Check that a window opens up in chrome that runs through the basic login tests, and they all pass
- Ignore the proxy request errors for now plz
- SHUT IT ALL DOWN
- In a single terminal run `yarn test-cypress`
- Check that the servers start up, cypress runs through the same passing tests, and then the process ends

Fixes #426 
Fixes #427
